test/ng/compileSpec.js

```diff
@@ -6126,154 +6126,114 @@ describe('$compile', function() {
-          var controllerCalled = false;
-          module(function($compileProvider, $controllerProvider) {
-            $controllerProvider.register('myCtrl', function() {
-              this.check = function() {
-                expect(this.data).toEqualData({
-                  'foo': 'bar',
-                  'baz': 'biz'
+                forEach([controllerOption, scopeOption, templateOption], function(option) {
+                  description.push(option.description);
+                  delete option.description;
+                  extend(ddo, option);
                 });
-                expect(this.data).toEqualData({
-                  'foo': 'bar',
-                  'baz': 'biz'
+
+                it('(' + description.join(', ') + ')', function() {
+                  var controllerCalled = false;
+                  module(function($compileProvider, $controllerProvider) {
+                    $controllerProvider.register('myCtrl', function() {
+                      this.check = function() {
+                        expect(this.data).toEqualData({
+                          'foo': 'bar',
+                          'baz': 'biz'
+                        });
+                        expect(this.oneway).toEqualData({
+                          'foo': 'bar',
+                          'baz': 'biz'
+                        });
+                        expect(this.str).toBe('Hello, world!');
+                        expect(this.fn()).toBe('called!');
+                      };
+                      controllerCalled = true;
+                      if (preAssignBindingsEnabled) {
+                        this.check();
+                      } else {
+                        this.$onInit = this.check;
+                      }
+                    });
+                    $compileProvider.directive('fooDir', valueFn(ddo));
+                  });
+                  inject(function($compile, $rootScope, $templateCache) {
+                    $templateCache.put('test.html', '<p>template</p>');
+                    $rootScope.fn = valueFn('called!');
+                    $rootScope.whom = 'world';
+                    $rootScope.remoteData = {
+                      'foo': 'bar',
+                      'baz': 'biz'
+                    };
+                    element = $compile('<div foo-dir dir-data="remoteData" ' +
+                                      'dir-str="Hello, {{whom}}!" ' +
+                                      'dir-fn="fn()"></div>')($rootScope);
+                    $rootScope.$digest();
+                    expect(controllerCalled).toBe(true);
+                    if (ddo.controllerAs || ddo.controller.indexOf(' as ') !== -1) {
+                      if (ddo.scope) {
+                        expect($rootScope.myCtrl).toBeUndefined();
+                      } else {
+                        // The controller identifier was added to the containing scope!
+                        // Should we allow this to happen?
```

Comment:
Why not? It's their code, they are allowed to do things :smiley:
In any case, this is not related to this PR. Remove this comment and feel free to open a new issue about it if you feel it shouldn't happen.